zk_toolbox/crates/zk_inception/src/commands/ecosystem/build.rs

```diff
@@ -0,0 +1,215 @@
+    forge_args: ForgeScriptArgs,
+    ecosystem_config: &EcosystemConfig,
+    initial_deployment_config: &InitialDeploymentConfig,
+    sender: String,
+) -> anyhow::Result<ContractsConfig> {
+    if ecosystem.build_ecosystem {
+        return build_ecosystem_inner(
+            shell,
+            forge_args,
+            ecosystem_config,
+            initial_deployment_config,
+            sender,
+        )
+        .await;
+    }
+
+    let ecosystem_preexisting_configs_path =
+        ecosystem_config
+            .get_preexisting_configs_path()
+            .join(format!(
+                "{}.yaml",
+                ecosystem_config.l1_network.to_string().to_lowercase()
+            ));
+
+    // currently there are not some preexisting ecosystem contracts in
+    // chains, so we need check if this file exists.
+    if ecosystem.ecosystem_contracts_path.is_none() && !ecosystem_preexisting_configs_path.exists()
+    {
+        anyhow::bail!(msg_ecosystem_no_found_preexisting_contract(
+            &ecosystem_config.l1_network.to_string()
+        ))
+    }
+
+    let ecosystem_contracts_path =
+        ecosystem
+            .ecosystem_contracts_path
+            .clone()
+            .unwrap_or_else(|| match ecosystem_config.l1_network {
+                L1Network::Localhost => {
+                    ContractsConfig::get_path_with_base_path(&ecosystem_config.config)
+                }
+                L1Network::Sepolia | L1Network::Holesky | L1Network::Mainnet => {
+                    ecosystem_preexisting_configs_path
+                }
+            });
+
+    ContractsConfig::read(shell, ecosystem_contracts_path)
+}
+
+async fn build_ecosystem_inner(
+    shell: &Shell,
+    forge_args: ForgeScriptArgs,
+    config: &EcosystemConfig,
+    initial_deployment_config: &InitialDeploymentConfig,
+    sender: String,
+) -> anyhow::Result<ContractsConfig> {
+    let deploy_config_path = DEPLOY_ECOSYSTEM_SCRIPT_PARAMS.input(&config.link_to_code);
+
+    let default_genesis_config =
+        GenesisConfig::read_with_base_path(shell, config.get_default_configs_path())
+            .context("Context")?;
+
+    let wallets_config = config.get_wallets()?;
+
+    // For deploying ecosystem we only need genesis batch params
+    let deploy_config = DeployL1Config::new(
+        &default_genesis_config,
+        &wallets_config,
+        initial_deployment_config,
+        config.era_chain_id,
+        config.prover_version == ProverMode::NoProofs,
+    );
+    deploy_config.save(shell, deploy_config_path)?;
+
+    let mut forge = Forge::new(&config.path_to_foundry())
+        .script(&DEPLOY_ECOSYSTEM_SCRIPT_PARAMS.script(), forge_args.clone())
+        .with_ffi()
+        .with_rpc_url("127.0.0.1:8545".to_string())
+        .with_sender(sender);
+
+    if config.l1_network == L1Network::Localhost {
+        // It's a kludge for reth, just because it doesn't behave properly with large amount of txs
+        forge = forge.with_slow();
+    }
```

Comment:
it's not needed for build

Reply:
Addressed in https://github.com/matter-labs/zksync-era/pull/2787/commits/4394d72e8873c0b884147bbb9af8c298454210b0